core/trino-main/src/main/java/io/trino/operator/exchange/LocalExchangeSourceOperator.java

```diff
@@ -48,7 +48,7 @@ public Operator createOperator(DriverContext driverContext)
             checkState(!closed, "Factory is already closed");
 
             OperatorContext operatorContext = driverContext.addOperatorContext(operatorId, planNodeId, LocalExchangeSourceOperator.class.getSimpleName());
-            return new LocalExchangeSourceOperator(operatorContext, localExchange.getNextSource());
+            return new LocalExchangeSourceOperator(operatorContext, localExchange.getNextSource(driverContext::getPhysicalWrittenDataSize));
```

Comment:
nit: comment in `DriverContext`:
```
Only calling getDriverStats is ThreadSafe
```
is not valid  as `getPhysicalWrittenDataSize` is also thread safe